src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -6744,25 +6746,19 @@ private static void loopChecks1a(int i, MethodHandle in, MethodHandle st) {
     }
 
     private static List<Class<?>> longestParameterList(Stream<MethodHandle> mhs, int skipSize) {
-        final List<Class<?>> empty = List.of();
-        final List<Class<?>> longest = mhs.filter(Objects::nonNull).
+        final Class<?>[] longest = mhs.filter(Objects::nonNull).
                 // take only those that can contribute to a common suffix because they are longer than the prefix
                         map(MethodHandle::type).
                         filter(t -> t.parameterCount() > skipSize).
-                        map(MethodType::parameterList).
-                        reduce((p, q) -> p.size() >= q.size() ? p : q).orElse(empty);
-        return longest.isEmpty() ? empty : longest.subList(skipSize, longest.size());
-    }
-
-    private static List<Class<?>> longestParameterList(List<List<Class<?>>> lists) {
-        final List<Class<?>> empty = List.of();
-        return lists.stream().reduce((p, q) -> p.size() >= q.size() ? p : q).orElse(empty);
+                        map(MethodType::ptypes).
+                        reduce((p, q) -> p.length >= q.length ? p : q).orElse(EMPTY);
```

Comment:
Could avoid the need to introduce `EMPTY` if you make the stream expression return the longest list directly:
```
                reduce((p, q) -> {
                    var longest = (p.size() >= q.size()) ? p : q;
                    return List.of(Arrays.copyOfRange(longest, skipSize, longest.size()); // checking isEmpty() is redundant here since we filter on `t.parameterCount() > skipSize`
                }).orElse(List.of());```

Reply:
Using lambdas inside MethodHandles is quite dangerous given that lambdas are initialized using method handles. It may work now because longuestParameterList() is not called when initializing a lambda but it may make any changes in the implementation of lambdas painfull in the future.

Reply:
Precious little method handle use in lambda bootstrap since JDK 11. Though I agree with the sentiment - having fixed a number of bootstrap issues in the past - `MethodHandles` is a small step up the abstraction ladder and the code in particular already uses a number of method refs and lambdas.

Reply:
ok, two small changes,
-  formatting: usually the method call in a stream are aligned with the '.' at the beginning
    ```
     stream
       .filter(...)
      .map(...)
   ```
   instead of at the end.

- the reduce is a max(),
  `max(Comparator.comparingInt(List::size))`

Reply:
@forax formatting is fixed. As of max() I think we can improve this even more by hoisting max() before calling ptypes()